components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/APIClient.java

```diff
@@ -147,11 +187,12 @@ private ActionInvocationResponse handleResponse(HttpResponse response) {
             case HttpStatus.SC_SERVICE_UNAVAILABLE:
             case HttpStatus.SC_GATEWAY_TIMEOUT:
                 actionInvocationResponseBuilder.errorLog(
-                        "Failed to execute the action request. Received: " + statusCode);
+                        "Failed to execute the action request for status code " + statusCode + ".");
                 actionInvocationResponseBuilder.retry(true);
                 break;
             default:
-                actionInvocationResponseBuilder.errorLog("Unexpected response status code: " + statusCode);
+                actionInvocationResponseBuilder.errorLog("Unexpected response received for status code " + statusCode
+                        + ".");
```

Comment:
Better say "Unexpected response received with status code {status}"

Reply:
Updated in